WcaOnRails/app/models/competition.rb

```diff
@@ -1,3 +1,189 @@
 class Competition < ActiveRecord::Base
   self.table_name = "Competitions"
+
+  has_many :registrations, foreign_key: "competitionId"
+  has_many :results, foreign_key: "competitionId"
+  has_many :competition_delegates
+  has_many :delegates, through: :competition_delegates
+  has_many :competition_organizers
+  has_many :organizers, through: :competition_organizers
+
+  ends_with_year_re = /\A.* \d{4}\z/
+  # TODO - learn ruby...
+  @@pattern_link_re = /\[\{([^}]+)}\{((https?:|mailto:)[^}]+)}\]/
+  pattern_text_with_links_re = /\A[^{}]*(#{@@pattern_link_re.source}[^{}]*)*\z/
+  validates :id, presence: true, uniqueness: true
+  validates :name, length: { maximum: 50 },
+                   format: { with: ends_with_year_re }
+  validates :cellName, length: { maximum: 45 },
+                       format: { with: ends_with_year_re }
+  validates :venue, format: { with: pattern_text_with_links_re }
+  validates :wcaDelegate, format: { with: pattern_text_with_links_re }
+  validates :organiser, format: { with: pattern_text_with_links_re }
+  validates :website, format: { with: pattern_text_with_links_re }
+
+  attr_accessor :start_date, :end_date
```

Comment:
Since you're overwriting the getters anyway, you could use `attr_writer :start_date, :end_date` which only generates `start_date=` and `end_date=`.